### PR TITLE
fix: respect authentication.skip.paths properly

### LINKED ISSUE #9206: Unable to skip basic authentication for specified paths in ksqldb-server
**Describe the bug**
Use of the config option `authentication.skip.paths` does not actually allow the specified paths to skip basic authentication for ksqldb-server.

**To Reproduce**
Steps to reproduce the behavior, include:
1. Using this [fork](https://github.com/mpistrang/cp-all-in-one/tree/ksqldb-server-basic-auth-example/cp-all-in-one-community) of [confluentinc/cp-all-in-one](https://github.com/confluentinc/cp-all-in-one)
2.  In the `cp-all-in-one-community` directory un `docker-compose -f docker-compose.yml -f docker-compose.ksqldb-server-basic-auth.yml up -d`
3. Note that in the `docker-compose.ksqldb-server-basic-auth.yml` override file basic auth is configured for ksqldb-server along with `/healthcheck` defined in the `authentication.skip.paths` option.

**Expected behavior**
The command `curl localhost:8088/healthcheck` with no basic auth provided returns the health check information

**Actual behavior**
The command `curl localhost:8088/healthcheck` returns an HTTP 401

**Additional context**
I expect there is additional configuration that I am missing here, but so far I have not found what it could be.

### PR DESCRIPTION
**Pull request by agavra**
### Description 
fixes #9206 by respecting the `authentication.skip.paths` in `KsqlAuthorizationProviderHandler.java`

### Testing done 
spun up CFK with MTLS and RBAC enabled on the normal endpoints. see relevant portion of configuration:
```
    authentication.skip.paths=/chc*,/heartbeat,/lag
    bootstrap.servers=kafka.operator-feyrhf.svc.cluster.local:9073
    config.providers=file
    config.providers.file.class=org.apache.kafka.common.config.provider.FileConfigProvider
    confluent.metadata.basic.auth.user.info=${file:/mnt/secrets/ksqldb-mds-client/bearer.txt:username}:${file:/mnt/secrets/ksqldb-mds-client/bearer.txt:password}
    confluent.metadata.bootstrap.server.urls=https://kafka.operator-feyrhf.svc.cluster.local:8090
    confluent.metadata.http.auth.credentials.provider=BASIC
    confluent.metadata.ssl.truststore.location=/mnt/sslcerts/truststore.jks
    confluent.metadata.ssl.truststore.password=${file:/mnt/sslcerts/jksPassword.txt:jksPassword}
    ksql.authentication.plugin.class=io.confluent.ksql.security.VertxBearerOrBasicAuthenticationPlugin
    ksql.heartbeat.enable=true
    ksql.lag.reporting.enable=true
    ksql.query.pull.enable.standby.reads=true
    ksql.security.extension.class=io.confluent.ksql.security.KsqlConfluentSecurityExtension
    ksql.streams.num.standby.replicas=1
    sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required metadataServerUrls="https://kafka.operator-feyrhf.svc.cluster.local:8090" username="${file:/mnt/secrets/ksqldb-mds-client/bearer.txt:username}" password="${file:/mnt/secrets/ksqldb-mds-client/bearer.txt:password}";
    sasl.login.callback.handler.class=io.confluent.kafka.clients.plugins.auth.token.TokenUserLoginCallbackHandler
    sasl.mechanism=OAUTHBEARER
    security.protocol=SASL_SSL
    ssl.enabled.protocols=TLSv1.2
    ssl.key.password=${file:/mnt/sslcerts/jksPassword.txt:jksPassword}
    ssl.keystore.location=/mnt/sslcerts/keystore.jks
    ssl.keystore.password=${file:/mnt/sslcerts/jksPassword.txt:jksPassword}
    ssl.truststore.location=/mnt/sslcerts/truststore.jks
    ssl.truststore.password=${file:/mnt/sslcerts/jksPassword.txt:jksPassword}
```

setting up unit testing for this is nigh impossible and the fix is blocking production users

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by nateab — APPROVED** [review]

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/KsqlAuthorizationProviderHandler.java [hunk 1]
```diff
@@ -33,19 +35,22 @@ public class KsqlAuthorizationProviderHandler implements Handler<RoutingContext>
...
+        server.getConfig().getList(KsqlRestConfig.AUTHENTICATION_SKIP_PATHS_CONFIG));
   }
 
   @Override
   public void handle(final RoutingContext routingContext) {
 
     final String path = routingContext.normalisedPath();
 
-    if (KSQL_AUTHENTICATION_SKIP_PATHS.contains(path)) {
+    if (unauthenticatedpaths.matcher(path).matches()) {
```
Comment (pgaref) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/KsqlAuthorizationProviderHandler.java:53]: Oh wow, so AUTHENTICATION_SKIP_PATHS_CONFIG was not taken into account at all before.

**Review by pgaref — APPROVED** [review]
Thanks for the FIX @agavra !
Changes LGTM

**Comment by pgaref** [comment]
This should also land to 7.0.x etc right?

**Comment by cadonna** [comment]
@agavra Thanks a lot for the fix! 
While I see that unit testing the whole setup is hard, I do not understand why unit testing `AuthenticationPluginHandler#getAuthorizationSkipPaths()` and `KsqlAuthorizationProviderHandler#handle()` should be hard. Could you clarify?

**Comment by cadonna** [comment]
> This should also land to 7.0.x etc right?

@pgaref Once this PR is merged to 6.0.x, we need to `pint merge` the commit so that it will bubble up to all `<major>.<minor>.x` branches including `7.0.x` and finally end up on master. See https://confluentinc.atlassian.net/wiki/spaces/QERM/pages/69009670/Release+Branches+and+Tags for more details about CPs branching strategy.

**Comment by agavra** [comment]
@cadonna there are no existing tests for these classes - I will add tests for `AuthenticationPluginHandler#getAuthorizationSkipPaths()` but tests for `KsqlAuthorizationProviderHandler#handle()` require a lot of mocking, and to be honest, won't add any meaningful coverage.

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/AuthenticationPluginHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  @Test
+  public void shouldBuildRegexThatRespectsSkipPaths() {
+    // Given:
+    final List<String> configs = ImmutableList.of("/heartbeat", "/lag");
+
+    // When:
+    final Pattern skips = AuthenticationPluginHandler.getAuthorizationSkipPaths(configs);
+
+    // Then:
+    assertThat(skips.matcher("/heartbeat").matches(), is(true));
```
Comment (cadonna) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/AuthenticationPluginHandlerTest.java:37]: Why not also verify for `/lags`? 
Could you also add a negative case, like `/shouldnotbefound`?

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/KsqlAuthorizationProviderHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  private Server server;
+  @Mock
+  private KsqlAuthorizationProvider authProvider;
+  @Mock
+  private WorkerExecutor workerExecutor;
+  @Mock
+  private RoutingContext routingContext;
+
+  @Test
+  public void shouldRespectServerAuthSkipPathsConfig() {
```
Comment (cadonna) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/KsqlAuthorizationProviderHandlerTest.java:44]: Could you also add a case where authentication is not skipped?

**Review by cadonna — COMMENTED** [review]
@agavra Thanks for the update!

I have two comments regarding the tests.

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/AuthenticationPluginHandlerTest.java (hunk 2 above)
Comment (agavra) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/AuthenticationPluginHandlerTest.java:37]: > Why not also verify for /lags?

that doesn't add any coverage, I'm ensuring that what is added in the config is able to skip

> Could you also add a negative case, like /shouldnotbefound?

Sure I'll add a negative test

Comment (agavra) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/AuthenticationPluginHandlerTest.java:37]: adding `/lag` in the config list makes sure that the regex works when you have multiple components - I'm not sure why checking that it works for `/lag` adds any coverage beyond checking for `/heartbeat` (basically it uses the same exact code path to check for `/lag` as it does to check for `/heartbeat`)

Comment (agavra) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/AuthenticationPluginHandlerTest.java:37]: I suppose I can see your point :) I'll add it in.

Comment (agavra) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/AuthenticationPluginHandlerTest.java:37]: https://github.com/confluentinc/ksql/pull/9229

**Pull request merged**